### PR TITLE
Add ADTimeDerivative

### PR DESCRIPTION
**Pull request by lindsayad**
Refs #5658 #12633

### CODE REVIEW

**Comment by lindsayad** [comment]
@tophmatthews

**Review by aeslaughter — DISMISSED** [review]

File: framework/src/kernels/ADTimeDerivative.C [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  : ADTimeKernel<compute_stage>(parameters)
+{
+}
+
+template <ComputeStage compute_stage>
+ADResidual
+ADTimeDerivative<compute_stage>::computeQpResidual()
+{
+  return _test[_i][_qp] * _u_dot[_qp];
+}
```
Comment (tophmatthews) [inline framework/src/kernels/ADTimeDerivative.C]: Doesn't this need something like `template class ADDiffusion<RESIDUAL>;` to inherit from this?

Comment (lindsayad) [reply framework/src/kernels/ADTimeDerivative.C]: Yep!

Comment (lindsayad) [reply framework/src/kernels/ADTimeDerivative.C]: Sigh, I just pushed haha.

Comment (tophmatthews) [reply framework/src/kernels/ADTimeDerivative.C]: Ah, alright. It was down to the line on that one!

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**